pkg/ybversion/version.go

```diff
@@ -0,0 +1,168 @@
+package ybversion
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type YBVersion struct {
```

Comment:
minor: can you add examples here?

for ex: 2.15.1.0-b1 translates to ...

also, do we handle pre release / custom versions like 2.15.1.0-b1-pre-release

Reply:
> minor: can you add examples here?

Will do.

> also, do we handle pre release / custom versions like 2.15.1.0-b1-pre-release

Not currently... Is how common is it to run across a version string like this in development environments?

Reply:
yeah, dev env builds and Jenkins builds for some pre-commit testing might show up this way but it is probably unlikely we should share those with customers, should be ok to skip.